code/modules/hydroponics/hydro_tray_process.dm

```diff
@@ -153,7 +153,7 @@
 	update_name() //fuck it i'll make it not happen constantly later
 
 	// Updates the plant overlay.
-	if(get_toxinlevel() > TOXINLEVEL_MAX/2)
+	if(get_toxinlevel() > get_waterlevel())
```

Comment:
I may just misunderstand this, but isn't this completely different?

Reply:
I missed something previously regarding the behavior, and this should function better for the overlay

Reply:
I don't think it should affect it too much

Reply:
anyway, I put the part back and it should work